packages/core/data-transfer/lib/engine/index.ts

```diff
@@ -260,11 +260,11 @@ class TransferEngine<
 
       // Gracefully close the providers
       await this.close();
-    } catch (e: any) {
-      throw e;
+    } catch (e: unknown) {
```

Comment:
Why do you type it as unknown? Shouldn't always be an instance of Error?

Reply:
Since the compiler can't be sure what is being thrown, catch will always be an unknown.

But this is an area that once we dive deeper into Typescript we should develop some standards around, such as having typeguards like [assertIsError](https://stackoverflow.com/questions/64452484/how-can-i-safely-access-caught-error-properties-in-typescript) that we always use.

Reply:
It could potentially be a string or smth else.
When we'll implement the rollback, we'll have to wrap its call with an `if e instanceof Error` or smth like that.
By the way, if you try to annotate the catch error type, you'll end up with the TS 1196 compiler error (which is "Catch clause variable type annotation must be 'any' or 'unknown' if specified.")

Reply:
I see, thanks!